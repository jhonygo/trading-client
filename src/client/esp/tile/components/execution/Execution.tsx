import React from 'react';
import { Transition } from 'react-spring';
import { ScaleLoader } from 'react-spinners';
import styled from 'styled-components';

const ExecutionStyle = styled.div`
  position: absolute;
  height: 100%;
  width: 100%;
  align-items: center;
  background-color: #3d4853;
`;

export namespace Execution {
  export interface Props {
    show: boolean;
  }
}

const Execution: React.FC<Execution.Props> = ({ show }) => (
  <Transition
    native
    reset
    unique
    items={0}
    from={{ opacity: 0 }}
    enter={{ opacity: 1 }}
    leave={{ opacity: 0 }}
  >
    {index =>
      show &&
      (style => (
        <ExecutionStyle style={style}>
          <ScaleLoader
            width={6}
            height={35}
            radius={2}
            color={'#3bafda;'}
            loading={true}
          />
        </ExecutionStyle>
      ))
    }
  </Transition>
);

export default Execution;
