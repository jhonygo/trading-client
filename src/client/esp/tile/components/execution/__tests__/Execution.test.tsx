import React from 'react';
import { shallow } from 'enzyme';
import Execution from '../Execution';

describe('Execution', () => {
  it('should display execution status', () => {
    // When
    const execution = shallow(<Execution show={true} />);

    // Then
    expect(execution).toMatchSnapshot();
  });

  it('should not display execution status', () => {
    // When
    const execution = shallow(<Execution show={false} />);

    // Then
    expect(execution).toMatchSnapshot();
  });
});
