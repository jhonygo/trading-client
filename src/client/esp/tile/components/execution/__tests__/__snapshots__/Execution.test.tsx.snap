// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Execution should display execution status 1`] = `
<Transition
  enter={
    Object {
      "opacity": 1,
    }
  }
  from={
    Object {
      "opacity": 0,
    }
  }
  items={0}
  keys={[Function]}
  leave={
    Object {
      "opacity": 0,
    }
  }
  native={true}
  reset={true}
  unique={true}
>
  <Component />
</Transition>
`;

exports[`Execution should not display execution status 1`] = `
<Transition
  enter={
    Object {
      "opacity": 1,
    }
  }
  from={
    Object {
      "opacity": 0,
    }
  }
  items={0}
  keys={[Function]}
  leave={
    Object {
      "opacity": 0,
    }
  }
  native={true}
  reset={true}
  unique={true}
>
  <Component />
</Transition>
`;
